Added Defold 1.12.3 CONSTANT_TYPE_TIME example (#175)
defold/examples: material/wave_background/example.md

---
tags: material
title: Wave Background

> brief: This example shows how to use Defold 1.12.3 CONSTANT_TYPE_TIME

Reviewer: `This example shows how to use Defold 1.12.3 CONSTANT_TYPE_TIME` -> `Shows how to use a Time shader constant to achieve a moving wave effect.` And in the description below, you can note it's using a time constant introduced in Defold 1.12.3, so that users now what version it works with.

> brief: This example shows how to use Defold 1.12.3 CONSTANT_TYPE_TIME

Reviewer: ```suggestion
brief: Shows how to use a Time shader constant to achieve a moving wave effect
```

author: JuLongZhiLu(巨龙之路), Pkeod(Brian Kramer)
scripts: wave_background.fp
thumbnail: thumbnail.png
---

> This example contains a game object with a sprite component. The Image and Default Animation properties of the sprite component cannot be left empty, otherwise an error will occur. I suggest directly using the built-in /builtins/graphics/particle_blob.tilesource and anim. You can adjust the size of the wave background by modifying the Size property of the sprite component.

Reviewer: Try to avoid personalized (first person) tone in example descriptions, instead use direct proposition/description:

`I suggest directly using` -> `In the example the built-in (...) is used.`

> This example contains a game object with a sprite component. The Image and Default Animation properties of the sprite component cannot be left empty, otherwise an error will occur. I suggest directly using the built-in /builtins/graphics/particle_blob.tilesource and anim. You can adjust the size of the wave background by modifying the Size property of the sprite component.

Reviewer: ```suggestion
This example contains a game object with a sprite component. The `Image` and `Default Animation properties` of the sprite component cannot be left empty, otherwise an error will occur. In the example the built-in `/builtins/graphics/particle_blob.tilesource` is used and animation is set to `anim`. You can adjust the size of the wave background by modifying the `Size` property of the sprite component.
```

> Related Defold forum post: https://forum.defold.com/t/open-source-defold-1-12-3-constant-type-time-example/82625

Reviewer: ```suggestion
Example uses a Fragment Constant of type `Time` introduced in Defold 1.12.3.
```
